lib/retriable.rb

```diff
@@ -59,13 +62,23 @@ def retriable(opts = {})
       begin
         return Timeout.timeout(timeout) { return yield(try) } if timeout
         return yield(try)
-      rescue *[*exception_list] => exception
+      rescue *[*rescue_list] => exception
         if on.is_a?(Hash)
           raise unless exception_list.any? do |e|
             exception.is_a?(e) && ([*on[e]].empty? || [*on[e]].any? { |pattern| exception.message =~ pattern })
           end
         end
 
+        if ignore.is_a?(Hash)
```

Comment:
Should this `ignore` check happen before the `on` check? Isn't it generally better if we raise as soon as we can since it'll shortcut checking the other exceptions?

Reply:
OK can do that.

Reply:
Done.